load_balancer/status.js

```diff
@@ -0,0 +1,90 @@
+/* eslint import/no-dynamic-require: 0 */
+const { Logger } = require('../util/logger.js');
+const { check } = require('../util/environmentCheck.js');
+const request = require('request');
+
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+/* The environment variable LBCONFIG will contain the path to the config file.
+  The actual path for the config is "../deploy/configs/load_balancer/nodes_data_conf.json"
+  For the docker containers, the path is /etc/load_balancer/nodes_data_conf.json */
+check('LBCONFIG');
+
+const lbConfig = require(process.env.LBCONFIG);
+
+const loggerConfig = lbConfig.load_balancer;
+loggerConfig.cmd = 'log';
+
+const logger = new Logger(loggerConfig);
+
+const getComponentResponse = function getComponentResponse(node, resolve) {
+  request.get(`https://${node.hostname}:${node.port}/connectionCheck`, (error, httpsResponse, body) => {
+    if (error) {
+      logger.error(`Error connecting to ${node.hostname}:${node.port}`);
+      logger.error(error);
+      resolve('down');
+    } else if (body.toString() === 'true') {
+      logger.debug(`Node at ${node.hostname}:${node.port} is up and running.`);
+      resolve('up');
```

Comment:
Is the name of the callback function consistent with the nomeclature of promises?

Reply:
Yes, the promises callback are resolve and reject.